src/components/pages/events/upcoming-events/upcoming-events.jsx

```diff
@@ -10,27 +10,39 @@ import styles from './upcoming-events.module.scss';
 
 const cx = classNames.bind(styles);
 
-const UpcomingEvents = ({ title, upcomingEvents }) => (
+const UpcomingEvents = ({ title, featuredUpcomingEvents, upcomingEventsByYear }) => (
   <section className={cx('wrapper')}>
     <div className="container">
       <Heading className={cx('title')}>{title}</Heading>
       <ul className={cx('items-wrapper')}>
         <div className={cx('cards-wrapper')}>
-          {upcomingEvents.slice(0, 3).map((upcomingEvent, index) => (
+          {featuredUpcomingEvents.map((upcomingEvent, index) => (
             <CardItem key={index} {...upcomingEvent} />
           ))}
         </div>
-        {upcomingEvents.slice(3).map((upcomingEvent, index) => (
-          <Item key={index} {...upcomingEvent} />
-        ))}
+        {upcomingEventsByYear && (
+        <div className={cx('events-wrapper')}>
+          {Object.keys(upcomingEventsByYear).map((year, index) => {
+            const events = Object.values(upcomingEventsByYear)[index];
```

Comment:
I think better option to go with `Object.entries` ([read more](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) here)